Prepend filename or identifier in HTML <title>

pcworld opened the pull request:
Display filenames and identifiers at the *beginning* of the <title>.
This way, they are better visible in modern browsers' tabs as they
usually cut off the title for space reasons.

This also fixes handling of the root path.
Examples of changed titles:
"Linux source code: (v5.5.6) - Bootlin" becomes "Linux source code (v5.5.6) - Bootlin"
"Linux source code: arch (v5.5.6) - Bootlin" becomes "arch - Linux source code (v5.5.6) - Bootlin"
"Linux source code: arch/um/Makefile (v5.5.6) - Bootlin" becomes "Makefile in arch/um - Linux source code (v5.5.6) - Bootlin"

---

Motivation: I usually have multiple tabs of Elixir open, and all I can read is "Linux source", which makes navigating the tabs really difficult.

michaelopdenacker commented:
Hi
Many thanks for this PR.

It definitely makes sense for browser tabs, but on the other hand, I still want the full file path in the title to make sure that we're going to match a search request.

Idents are fine in your proposed changes.
For file names, what about something like this ?

sys_titan.c - arch/alpha/kernel/sys_titan.c - Linux source code (v5.5.6) - Bootlin

That's longer but may address both needs: having a useful browser tab title and the full patch for matching in search engines

Any thoughts? @carmeli-tamir @cxw42

cxw42 commented:
I agree with `file - full_path - project+version`.  I looked at the code and don't have any particular concerns.  Thanks!

carmeli-tamir commented on http/web.py:
```diff
@@ -174,7 +175,10 @@ def print(arg, end='\n'):
     data['ident'] = ident
-    data['title'] = project.capitalize()+' source code: '+path[1:]+' ('+tag+') - Bootlin'
+    data['title'] = ('' if path == ''
+                     else path_split[1]+' - ' if len(path_split) == 2
```
1. Since p3 is already defined, you can use `p3[0]` instead of `path_split[1]`, `len(p3) == 1` instead of `len(path_split) == 2`, `p3[:-1]` instead of `path_split[1:-1]` etc.
2. Maybe rename p3 to `name_split` and remove the current `name_split` variable which is only used to create p3.

pcworld replied:
Done. I've further renamed `p2` to `path_temp`.

carmeli-tamir commented on http/web.py:
```diff
@@ -277,7 +281,7 @@ def print(arg, end='\n'):
 
 elif mode == 'ident':
-    data['title'] = project.capitalize()+' source code: '+ident+' identifier ('+tag+') - Bootlin'
+    data['title'] = ident+' identifier - '+project.capitalize()+' source code ('+tag+') - Bootlin'
```
In order to save some code dup, I suggest to create a variable (e.g. title_base_name) whose value is `project.capitalize()+' source code ('+tag+') - Bootlin'` and use it here and in line 181

pcworld replied:
Done.

carmeli-tamir commented:
Commented with minor suggestions

pcworld commented:
I've implemented the suggested changes.
Further I've added a comment to the code with example paths, because otherwise the code's purpose might be difficult to understand.

cxw42 commented:
LGTM.  (Note, though, that I haven't tested it myself :) .)  Thanks, @pcworld!

michaelopdenacker commented:
Well done, you anticipated every possible case.
Accepting the change, many thanks :)
Michael.
